solr/webapp/web/js/angular/controllers/analysis.js

```diff
@@ -21,33 +21,64 @@ solrAdminApp.controller('AnalysisController',
 
       $scope.refresh = function() {
         Luke.schema({core: $routeParams.core}, function(data) {
-          $scope.fieldsAndTypes = [];
-          for (var field in data.schema.fields) {
-            $scope.fieldsAndTypes.push({
-              group: "Fields",
-              value: "fieldname=" + field,
-              label: field});
-          }
-          for (var type in data.schema.types) {
-            $scope.fieldsAndTypes.push({
-              group: "Types",
-              value: "fieldtype=" + type,
-              label: type});
-          }
+          $scope.fieldsAndTypes = getFieldsAndTypes(data.schema);
           $scope.core = $routeParams.core;
+          $scope.parseQueryString(data.schema);
+          // @todo - set defaultSearchField either to context["analysis.fieldname"] or context["analysis.fieldtype"]
```

Comment:
is this still a valid todo?

Reply:
I don't see defaultSearchField used in the JS or HTML?

Reply:
I am not exactly sure, left the todo there just in case

Reply:
Cool, what prompted this?     I guess I am wondering if we need this?   I am playing with the code, and it *appears* to all work ;-)